cedar-lean/Cedar/Thm/Validation.lean

```diff
@@ -0,0 +1,46 @@
+import Cedar.Spec
+import Cedar.Data
+import Cedar.Validation
+import Cedar.Thm.Validation.Validator
+
+namespace Cedar.Thm
+
+open Cedar.Data
+open Cedar.Spec
+open Cedar.Validation
+
+/--
+If validation succeeds, then for any request consistent with the schema, either
+(1) evaluation of every policy produces a boolean or (2) returns an error TODO
+-/
+theorem validation_is_sound (policies : Policies) (schema : Schema) (request : Request) (entities : Entities) :
+validate policies schema = .ok () →
+(∀ env ∈ schema.toEnvironments,
+RequestAndEntitiesMatchEnvironment env request entities) →
+AllEvaluateCorrectly policies request entities := by
+intro h₀ h₁
+unfold validate at h₀
+simp [AllEvaluateCorrectly]
+cases h₃ : policies with
+| nil => simp [h₃]
+| cons h' t' =>
+  intro policy pin
+  simp [OneEvaluatesCorrectly]
```

Comment:
We follow Mathlib guidelines and avoid using `simp [...]` except to close a goal.  This helps with proof stability. 

We use `simp only [...]` instead.  If you replace a given occurrence of `simp [...]` with `simp? [...]`, Lean will tell you how to rewrite it into `simp only [...]`.